src/btreemap/node.rs

```diff
@@ -65,13 +66,24 @@ pub struct Node<K: Storable + Ord + Clone> {
 
 impl<K: Storable + Ord + Clone> Node<K> {
     /// Loads a node from memory at the given address.
-    pub fn load<M: Memory>(address: Address, memory: &M, version: Version) -> Self {
-        match version {
-            Version::V1(DerivedPageSize {
-                max_key_size,
-                max_value_size,
-            }) => Self::load_v1(address, max_key_size, max_value_size, memory),
-            Version::V2(page_size) => Self::load_v2(address, page_size, memory),
+    pub fn load<M: Memory>(address: Address, page_size: PageSize, memory: &M) -> Self {
+        // Load the header to determine which version the node is, then load the node accordingly.
+        let header: NodeHeader = read_struct(address, memory);
+        assert_eq!(&header.magic, MAGIC, "Bad magic.");
+        match header.version {
+            LAYOUT_VERSION_1 => {
+                if let PageSize::Derived(DerivedPageSize {
+                    max_key_size,
+                    max_value_size,
+                }) = page_size
+                {
+                    Self::load_v1(header, address, max_key_size, max_value_size, memory)
+                } else {
+                    unreachable!("Tried to load a V1 node without a derived PageSize.")
+                }
```

Comment:
nit: Use a match since we want to handle both cases? It's a bit more idiomatic and we prevent accidentally not handling some case if we add more in the future.

Reply:
Done.